### ISSUE TITLE
Use torch DataLoader if torchtupeles DataLoaderSlice is not supported

### ISSUE THREAD

**Issue by havakv** [havakv/torchtuples#8]
For versions of torch that cannot take advantage of `torchtuples.data.DataLoaderSlice` we should default to `torch.utils.data.DataLoader` with a warning instead of raising an error.
